pkg/apis/provisioning/v1alpha1/provisioner.go

```diff
@@ -15,14 +15,15 @@ limitations under the License.
 package v1alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type ProvisionerSpec struct {
 	// +optional
-	Cluster     *ClusterSpec    `json:"cluster,omitempty"`
-	Allocator   AllocatorSpec   `json:"allocator,omitempty"`
-	Reallocator ReallocatorSpec `json:"reallocator,omitempty"`
+	Cluster      *ClusterSpec     `json:"cluster,omitempty"`
+	Allocation   AllocationSpec   `json:"allocation,omitempty"`
+	Reallocation ReallocationSpec `json:"reallocation,omitempty"`
```

Comment:
I am personally a fan of Allocator/Reallocator as they are controllers doing some action. Allocation/Reallocation sounds like a one-off action to me. If you rename to Allocation/Reallocation, I think it would make sense to also go with ProvisioningSpec.

Reply:
Maybe we can discuss as a group? I made this change because I felt that the "thing-er" pattern exposed too many implementation details. For example, what if allocation and reallocation were implemented by a single process, instead of two. By stating that we have an allocator rather than an allocation policy, we expose the detail of that process. 

I'm a little on the fence with this, as I couldn't find any great examples other than the lack of thing-ers across the kubernetes ecosystem (e.g. Deployment vs Deployer). Happy to hear any arguments you have.

Reply:
Gotcha. I think that separating into the words allocation and reallocation as opposed to just provisioning always implies that there's two processes going on, but I do agree with keeping in line with the kubernetes naming scheme. We can punt on this for another refactor later.